src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp

```diff
@@ -244,6 +211,12 @@ namespace model {
         result.push_back(curve.get());
       }
 
+      if (auto const _loadingIndexList = loadingIndexList()) {
+        for (const auto& mo : _loadingIndexList->modelObjects()) {
+          result.push_back(mo);
+        }
+      }
```

Comment:
pedantic man to the rescue!

```suggestion
      if (auto const _loadingIndexList = loadingIndexList()) {
        auto mos =  _loadingIndexList->modelObjects();
        result.reserve(result.size() + mos.size());
        result.insert(result.end(), std::make_move_iterator(mos.begin()), std::make_move_iterator(mos.end()));
      }
```

Anyways, what makes me tick is the fact that `_loadingIndexList` is an optional, it shouldn't be. An please use a trailing optional_, not _optional, technically it's UB as anything starting with an underscore is reserved for the STL implementators. (I am probably the main person to blame for this pattern of a leading underscore for optional...)

Reply:
I took this straight from `CoilHeatingDXVariableSpeed.cpp`...

Reply:
hehe. Well the Ctor there does create the MoList... I think making non optional + OS_ASSERT when fetching it makes more sense. As far as make_move_iterator, it's just a side note, ignore it.